service/shutdown.go

```diff
@@ -40,17 +40,19 @@ func (s *ShutdownHandler) Add(service Shutdowner) {
 func (s *ShutdownHandler) Wait(ctx context.Context) error {
 	<-ctx.Done()
 
-	p := pool.NewWithResults[error]()
+	errs := make([]error, len(s.services))
+	g := &errgroup.Group{}
 
-	for _, v := range s.services {
-		service := v
-
-		p.Go(func() error {
+	for i, service := range s.services {
+		g.Go(func() error {
```

Comment:
Go 1.22 fix of the for-loop variable is such a peace of mind

Reply:
Indeed